tests/linux_kernel/helpers/test_waitq.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+from drgn import Object
+from drgn.helpers.linux.waitq import (
+    for_each_entry_in_waitqueue,
+    for_each_task_in_waitqueue,
+    waitqueue_active,
+)
+from tests.linux_kernel import LinuxKernelTestCase, skip_unless_have_test_kmod
+
+
+def waiter_tasks(prog):
+    return [
+        prog["drgn_test_waitq_kthread"],
+    ]
+
+
+@skip_unless_have_test_kmod
+class TestWaitqueue(LinuxKernelTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.waitq = cls.prog["drgn_test_waitq"].address_of_()
+        cls.empty_waitq = cls.prog["drgn_test_empty_waitq"].address_of_()
+        cls.kthread = cls.prog["drgn_test_waitq_kthread"]
+
+    def test_waitqueue_active(self):
+        self.assertTrue(waitqueue_active(self.waitq))
+        self.assertFalse(waitqueue_active(self.empty_waitq))
+
+    def test_for_each_entry_in_waitqueue(self):
+        self.assertEqual(
+            [entry for entry in for_each_entry_in_waitqueue(self.empty_waitq)],
+            [],
+        )
+        self.assertEqual(
+            [
+                Object(
+                    self.prog, "struct task_struct", address=entry.private.value_()
+                ).address_of_()
+                for entry in for_each_entry_in_waitqueue(self.waitq)
+            ],
+            waiter_tasks(self.prog),
+        )
```

Comment:
This test doesn't seem that valuable to me since it's basically reimplementing `for_each_task_in_waitqueue()`. I'd say that we either make the `wait_queue_entry_t` in the test kmod global (e.g., by open-coding `wait_event_interruptible()`)  so that we can compare it directly here or just drop this.

Reply:
I think we can drop this safely because waitqueue_for_each_task would anyways be testing waitqueue_for_each_entry as well.